### PR TITLE
IAzorius documentation

### PR DESCRIPTION
**Pull request by herbig**
## Description

Fleshing out documentation for the IAzorius interface.

### CODE REVIEW

File: contracts/azorius/interfaces/IAzorius.sol [hunk 1]
```diff
@@ -3,22 +3,65 @@ pragma solidity ^0.8.0;
...
+        // hashes of the transactions that are being proposed
+        bytes32[] txHashes;
+        // time (in seconds) this proposal will be timelocked for if it passes
+        uint256 timelockPeriod;
+        // time (in seconds) this proposal has to be executed after timelock
+        // ends before it is expired
+        uint256 executionPeriod;
+        // count of transactions that have been executed within the proposal
+        uint256 executionCounter;
     }
```
[the remark this answers is not in the document]
Comment (herbig) [reply contracts/azorius/interfaces/IAzorius.sol:41]: yeah fair, I'll update

File: contracts/azorius/interfaces/IAzorius.sol [hunk 2]
```diff
@@ -28,38 +71,59 @@ interface IAzorius {
...
 
-    /// @notice Executes the specified transaction within a proposal
-    /// @notice Transactions must be called in order
-    /// @param _proposalId the identifier of the proposal
-    /// @param _target the contract to be called by the avatar
-    /// @param _value ether value to pass with the call
-    /// @param _data the data to be executed from the call
-    /// @param _operation Call or Delegatecall
+    /**
+     * Executes the specified Proposal.
```
[the remark this answers is not in the document]
Comment (herbig) [reply contracts/azorius/interfaces/IAzorius.sol:117]: yeah good call, I'll add more context around that, now that we're making it private as well

**Comment by herbig** [comment]
> Looks good! Main thing I'm worried about is removing the natspec from the Azorius contract. Why would we not just want to include the documentation in both the contract and the interface?

@tbwebb22 having the same comments repeated in two locations makes it likely they'll get out of sync, or at the least require updating two places at once to change, rather than having the implementation just say go look at the interface

File: contracts/azorius/Azorius.sol [hunk 3]
```diff
@@ -168,13 +158,16 @@ contract Azorius is Module, IAzorius {
...
+    /**
+     * Executes the specified transaction in a Proposal, by index.
+     * Transactions in a proposal must be called in order.
+     *
+     * @param _proposalId identifier of the proposal
+     * @param _target contract to be called by the avatar
+     * @param _value ETH value to pass with the call
+     * @param _data data to be executed from the call
+     * @param _operation Call or Delegatecall
+     */
```
Comment (sethhrbek) [inline contracts/azorius/Azorius.sol:170]: would a more accurate name be `executeProposalTxByIndex`?

File: contracts/azorius/interfaces/IAzorius.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (sethhrbek) [reply contracts/azorius/interfaces/IAzorius.sol]: left a comment about potentially updating the name of the function as well

File: contracts/azorius/Azorius.sol (hunk 3 above)
Comment (tbwebb22) [reply contracts/azorius/Azorius.sol:170]: Yeah, agreed I think that's a better description

**Comment by tbwebb22** [comment]
> 

Ok, I'm cool with this. I dug into it a bit more, didn't know that the Natspec was also inherited from base functions: 
https://docs.soliditylang.org/en/v0.8.17/natspec-format.html#inheritance-notes

Also didn't know about that `@inheritdoc` tag which is good to know

File: contracts/azorius/Azorius.sol (hunk 3 above)
Comment (tbwebb22) [reply contracts/azorius/Azorius.sol:170]: @herbig Do you want to address this function name change in this PR, or would you like me to change it my current branch?

**Comment by herbig** [comment]
> > 
> 
> Ok, I'm cool with this. I dug into it a bit more, didn't know that the Natspec was also inherited from base functions: https://docs.soliditylang.org/en/v0.8.17/natspec-format.html#inheritance-notes
> 
> Also didn't know about that `@inheritdoc` tag which is good to know

oh cool yeah that's good to know, I should remove all the docs on the Azorius contract that are implementations of the interface, or add that tag

File: contracts/azorius/Azorius.sol (hunk 3 above)
Comment (herbig) [reply contracts/azorius/Azorius.sol:170]: I can update this, no prob

Comment (tbwebb22) [reply contracts/azorius/Azorius.sol:170]: Oh yeah, also making this private will cause some tests to fail. I can push commit to fix those on this branch unless you want to tackle it

Comment (herbig) [reply contracts/azorius/Azorius.sol:170]: Ah true, yeah might be quicker for you to fix those: https://github.com/decent-dao/fractal-contracts/pull/35/commits/662162003dd69fa264b3d74204d79de8ceb52c01

Comment (tbwebb22) [reply contracts/azorius/Azorius.sol:170]: This is done

**Review by tbwebb22 — APPROVED** [review]

**Review by sethhrbek — APPROVED** [review]

**Pull request merged**